src/Data/ByteString/Base16/Internal/W32/Loop.hs

```diff
@@ -47,22 +49,9 @@ innerLoop !dptr !sptr !end = go dptr sptr
 
     !alphabet = "0123456789abcdef"#
 
-    tailRound16 !dst !src
-      | src == end = return ()
-      | otherwise = do
-        !t <- peek @Word8 src
-
-        let !a = fromIntegral (lix (unsafeShiftR t 4))
-            !b = fromIntegral (lix t)
-
-        let !w = a .|. (unsafeShiftL b 8)
-
-        poke @Word16 dst w
-
-        tailRound16 (plusPtr dst 2) (plusPtr src 1)
-
     go !dst !src
-      | plusPtr src 3 >= end = tailRound16 (castPtr dst) (castPtr src)
```

Comment:
Removed the duplicate code and just called out to `W16.innerLoop` and `W32.innerLoop` below.